src/devices/bus/pce_ctrl/joypad2.cpp

```diff
@@ -0,0 +1,242 @@
+// license:BSD-3-Clause
+// copyright-holders:cam900
+/**********************************************************************
+
+    NEC PC Engine/TurboGrafx-16 2 button joypad emulation
+
+    Based on SMS controller port emulation (devices\bus\sms_ctrl\*.*)
+    by Fabio Priuli,
+    PC engine emulation (mame\*\pce.*)
+    by Charles MacDonald, Wilbert Pol, Angelo Salese
+
+    First party models:
+
+    NEC PC Engine Pad (PI-PD001)
```

Comment:
> I agree with your comments, except this one:
> 
> > * All the controllers have a “turbo” feature for the I and II buttons (two-speed for the standard pad and Avenue Pad 3, single-speed for the Avenue Pad 6) driven off the counter.
> 
> Not all of them do, although most do. The controllers initially packaged and sold with the original PC Engine did not have the turbo feature, although turbo controllers were available soon afterward.

That's are also emulated and documented in joypad2.cpp, It named as just "PC-Engine Pad", PI-PD001 model number.